### PR TITLE
chore(utils): remove type errors

### PR DESCRIPTION
**Pull request by eyllanesc**

### CODE REVIEW

File: src/poetry/utils/exporter.py [hunk 1]
```diff
@@ -174,9 +177,11 @@ def _export_requirements_txt(
...
-            output.write(decoded)
-        except AttributeError:
+        if isinstance(output, str):
             filepath = cwd / output
             with filepath.open("w", encoding="utf-8") as f:
                 f.write(decoded)
+            return
+        with contextlib.suppress(AttributeError):
+            # FIXME
+            output.write(decoded)
```
Comment (branchvincent) [inline src/poetry/utils/exporter.py:187]: I imagine we don't need to worry about `AttributeError` and can just rely on the instance check:

```suggestion
        output.write(decoded)
```

**Comment by dimbleby** [comment]
#5279

**Comment by dimbleby** [comment]
I seem to have largely duplicated this in #5464, sorry about that.  It's a shame both because of the unnecessary extra work but also because there are things here that I probably like more than what I did, in particular the extra types you added look interesting.

**Comment by Secrus** [comment]
This PR has already been addressed by other contributions and is no longer valid.

**Pull request closed without merging**